src/meter_recognition/fit_to_bar.py

```diff
@@ -0,0 +1,117 @@
+from librosa import util
+from utils import constants
+
+
+def beat_id_closest_to_timestamp(beats, timestamp):
+    clos = 1000
+    res = -1
+    for i in range(len(beats)):
+        if abs(beats[i]-timestamp) < clos:
+            clos = abs(beats[i]-timestamp)
+            res = i
+
+    return res
+
+
+def simple_sound_beat_dur(beats, sound):
+    id = sound.beat_id
+    b1 = beats[id]
+    b2 = beats[id+1]
+    sixteenth = (b2-b1)/4
```

Comment:
By the way, I would write this function like that:
`local_beat_duration=beats[sound.beat_id+1]-beats[sound.beat_id]
local_sixteenth_duration = local_beat_duration/4
no_sixteenths_in_sound = sound.duration_ms/local_sixteenth_duration
return no_sixteenths_in_sound'
What do you think? For me it's much easier to read and understand

Reply:
4/4 and 4/8 for the sake of algorithm are the same. There could be a problem with 3/4 or something like this

Reply:
Wait a second, is one beat a whole polish "takt", or just one accent (for example 4 beats in "takt" in 4/4)? Bc I assumed the second version and then 4/4 and 3/4 are the same - beat is a one fourth note, there are simply four or three 4ths in "takt", but it doesn't matter. 4/8 would be different, bc then beat=8th note, beat/4=32th note, not 16th

Reply:
one beat is just one accent
beat is a one fourth note
my previous comment didn't have any sense
4/8 would not be different because it would just imact notation, we can scale it up or down later for better user experience. Morover there is no way of detecting 4/8 vs 4/4 vs 4/16

Reply:
:D it's ok with this comment now